File: lesson-0-7-javascript-foundations-for-qa/service_checker.js
const services = ["api", "database", "cache", "search", "auth"];

const results = {
    api: "UP",
    database: "UP",
    cache: "UP",
    search: "UP",
    auth: "UP"
};

function printHealthReport(results) {
    console.log("=== Service Health Report ===");

    let upCount = 0;
    let downCount = 0;

    for (const service of services) {
        const status = results[service];

        console.log(service + ": " + status);

        if (status === "UP") {
            upCount++;
        } else if (status === "DOWN") {
            downCount++;
        }
    }

    console.log("");
    console.log("Up: " + upCount);
    console.log("Down: " + downCount);
    console.log("Total: " + services.length);
}

printHealthReport(results);

File: lesson-0-6-python-foundations-for-qa/service_checker.py
services = ["api", "database", "cache", "search", "auth"]

results = {}

for service in services:
    answer = input(f"Is {service} up? (yes/no): ").strip().lower()

    if answer == "yes":
        results[service] = "UP"
    else:
        results[service] = "DOWN"


def print_health_report(results):
    print("=== Service Health Report ===")

    for service, status in results.items():
        print(f"{service}: {status}")

    up_count = 0
    down_count = 0

    for status in results.values():
        if status == "UP":
            up_count += 1
        else:
            down_count += 1

    print()
    print(f"Up: {up_count}")
    print(f"Down: {down_count}")
    print("Total: 5")


print_health_report(results)